seccomp.c

```diff
@@ -47,17 +47,41 @@ FILE* sc_must_read_and_validate_header_from_file(const char *profile_path, struc
 void sc_must_read_filter_from_file(FILE *file, uint32_t len_bytes, struct sock_fprog *prog)
 {
 	prog->len = len_bytes / sizeof(struct sock_filter);
-	prog->filter = (struct sock_filter *)malloc(MAX_BPF_SIZE);
+	// When reading syscall numbers or other data, ensure they are within valid ranges. This should be done in the loop where the file is being read:
+	if (prog->len > MAX_BPF_SIZE / sizeof(struct sock_filter)) {
+	    die("seccomp filter too large");
```

Comment:
`die()` effectively closes the whole program, so it is no return. No use of calling stuff after `die()`.